File: 1013-fibonacci-number/1013-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        vector<long long> dp;
        dp.push_back(0);
        dp.push_back(1);
        if (n <= 1) {
            return n;
        }
        for (int i = 2; i <= n; ++i) {
            dp.push_back(dp[i - 1] + dp[i - 2]);
        }
        return dp[n];
    }
};

File: 1013-fibonacci-number/1013-fibonacci-number.py
class Solution:
    dp = dict()
    def fib(self, n: int) -> int:
        if n in self.dp: return self.dp[n]
        if n <= 1: return n
        self.dp[n] = self.fib(n - 1) + self.fib(n - 2)
        return self.dp[n]